lib/services/database/sqlite-database.dart

```diff
@@ -98,270 +96,326 @@ class SqliteDatabase implements DatabaseInterface {
-            return Record.fromMap(currentRowMap);
-        });
-        return (matching.isEmpty) ? null : matching.first;
     }
-
-    @override
-    Future<List<Record>> getAllRecords() async {
-        final db = (await database)!;
-        var maps = await db.rawQuery("""
+    var matching = List.generate(maps.length, (i) {
+      Map<String, dynamic> currentRowMap = Map<String, dynamic>.from(maps[i]);
+      currentRowMap["category"] = Category.fromMap(currentRowMap);
+      return Record.fromMap(currentRowMap);
+    });
+    return (matching.isEmpty) ? null : matching.first;
+  }
+
+  @override
+  Future<List<Record>> getAllRecords() async {
+    final db = (await database)!;
+    var maps = await db.rawQuery("""
             SELECT m.*, c.name, c.color, c.category_type, c.icon
             FROM records as m LEFT JOIN categories as c ON m.category_name = c.name AND m.category_type = c.category_type
         """);
-        return List.generate(maps.length, (i) {
-            Map<String, dynamic> currentRowMap = Map<String, dynamic>.from(maps[i]);
-            currentRowMap["category"] = Category.fromMap(currentRowMap);
-            return Record.fromMap(currentRowMap);
-        });
-    }
-
-    @override
-    Future<List<Record>> getAllRecordsInInterval(DateTime? from, DateTime? to) async {
-        final db = (await database)!;
-        final fromUnix = from!.millisecondsSinceEpoch;
-        final toUnix = to!.millisecondsSinceEpoch;
-
-        var maps = await db.rawQuery("""
+    return List.generate(maps.length, (i) {
+      Map<String, dynamic> currentRowMap = Map<String, dynamic>.from(maps[i]);
+      currentRowMap["category"] = Category.fromMap(currentRowMap);
+      return Record.fromMap(currentRowMap);
+    });
+  }
+
+  @override
+  Future<List<Record?>> searchRecordsInCategory(
+      String search, String categoryName) async {
+    final db = (await database)!;
+    var maps = await db.rawQuery("""
+            SELECT m.*, c.name, c.color, c.category_type, c.icon
+            FROM records as m LEFT JOIN categories as c ON m.category_name = c.name AND m.category_type = c.category_type WHERE m.title LIKE ? AND c.name = ?
+        """, ["%$search%", categoryName]);
+    return List.generate(maps.length, (i) {
+      Map<String, dynamic> currentRowMap = Map<String, dynamic>.from(maps[i]);
+      currentRowMap["category"] = Category.fromMap(currentRowMap);
+      return Record.fromMap(currentRowMap);
+    });
+  }
+
+  @override
+  Future<List<String>> suggestedRecordNames(
+      String search, String categoryName) async {
+    final db = (await database)!;
+    var maps = await db.rawQuery("""
+            SELECT DISTINCT m.title 
+            FROM records as m WHERE m.title LIKE ? AND m.category_name = ? 
+        """, ["%$search%", categoryName]);
+    return List.generate(maps.length, (i) {
+      Map<String, dynamic> currentRowMap = Map<String, String>.from(maps[i]);
+      return currentRowMap["title"];
+    });
+  }
```

Comment:
Which one do you like more?

Reply:
Difficult choice indeed. I would say the second approach as it fits more the current use-case. The first might be used for other functionalities as well (maybe implementing a search function?) but since the requirements are not clear yet, it does not make much sense.